src/Relations/BelongsToMany.php

```diff
@@ -126,12 +126,7 @@ public function sync($ids, $detaching = true)
         // if they exist in the array of current ones, and if not we will insert.
         $current = $this->parent->{$this->relatedPivotKey} ?: [];
 
-        // See issue #256.
-        if ($current instanceof Collection) {
-            $current = $ids->modelKeys();
-        }
```

Comment:
Could you please remove these lines in the[ `MorphToMany`](https://github.com/hans-thomas/laravel-mongodb/blob/d47380ccfb3cba967855191600847133726ceed3/src/Relations/MorphToMany.php#L136-L139) class too?